### PR TITLE
Update repo .vsconfig

### LINKED ISSUE #13328: Update repo .vsconfig
The repo `.vsconfig` can be imported in the Visual Studio Installer to ensure that all the VS components needed to build the repo, and run tests, are installed.  Note, Apex/E2E tests additionally need UWP components, which are not included, the `.vsconfig` is just enough to run CLI tests.

### PR DESCRIPTION
**Pull request by zivkan**
## Bug

Fixes: https://github.com/NuGet/Home/issues/13328

Regression? no

## Description

Update `.vsconfig` so that the VS Installer doesn't show an error when importing, and also installed all the components needed to run CLI tests.

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [x] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

File: .vsconfig [hunk 1]
```diff
@@ -1,25 +1,34 @@
...
     "Microsoft.VisualStudio.Component.ManagedDesktop.Core",
+    "Microsoft.NetCore.Component.Runtime.8.0",
+    "Microsoft.NetCore.Component.SDK",
+    "Microsoft.VisualStudio.Component.FSharp",
+    "Microsoft.ComponentGroup.ClickOnce.Publish",
+    "Microsoft.NetCore.Component.DevelopmentTools",
     "Microsoft.VisualStudio.Component.IntelliCode",
     "Microsoft.VisualStudio.Component.ManagedDesktop.Prerequisites",
     "Microsoft.VisualStudio.Workload.ManagedDesktop",
+    "Microsoft.Net.Component.4.6.TargetingPack",
```
Comment (nkolev92) [inline .vsconfig:25]: Tests?

**Review by nkolev92 — APPROVED** [review]

File: .vsconfig (hunk 1 above)
Comment (zivkan) [reply .vsconfig:25]: No. The Visual Studio SDK lists it as a dependency, so trying to uninstall it from the VS Installer will also uninstall the VS SDK.

**Review by jeffkl — APPROVED** [review]

File: .vsconfig [hunk 2]
```diff
@@ -1,25 +1,34 @@
...
     "Microsoft.VisualStudio.Component.VSSDK",
     "Microsoft.VisualStudio.ComponentGroup.VisualStudioExtension.Prerequisites",
     "Microsoft.VisualStudio.Workload.VisualStudioExtension",
-    "Microsoft.VisualStudio.Workload.NetCoreTools"
-  ]
-}
+    "Microsoft.NetCore.Component.Runtime.3.1",
+    "Microsoft.NetCore.Component.Runtime.5.0",
```
Comment (kartheekp-ms) [inline .vsconfig:30]: Is it okay to install these runtimes (3.1 and 5.0) even though the corresponding .NET SDK version is out of support?

Comment (zivkan) [reply .vsconfig:30]: Our tests use them, so the tests won't run without them. Doesn't mean it's a good idea for us to use it for tests, but I just set up a new machine and wanted to share an updated .vsconfig that actually works.

**Pull request merged**